### DISCUSSION TITLE
I dont know why ：No module name''setuptools.py27compat

### DISCUSSION THREAD

**Discussion by itues** [pyinstaller/pyinstaller#5975] [Help]
Output of pyinstaller --version: 4.3

Version of Python: 3.6.2

Platform: Windows10

My  English is poor，bug as down picture，please have a c c，thanks

![image](https://user-images.githubusercontent.com/86935392/124539164-83cd8900-de4f-11eb-9329-24efcd5fb0c9.png)

**Comment by rokm** [comment]
You are getting this error because you are using a relatively old version of `setuptools` that still has `py27compat` (and `py33compat`) modules, and because [PyInstaller's `setuptools` hook explicitly excludes these two modules](https://github.com/pyinstaller/pyinstaller/blob/ab81fe39b11bc12216f86e85fb899ff13379c069/PyInstaller/hooks/hook-setuptools.py#L16).

You can either:
* explicitly add those modules to `hiddenimports`, for example by adding the following to your PyInstaller command-line (provided you are not building from .spec): `--hiddenimport setuptools.py33compat --hiddenimport setuptools.py27compat`
* upgrade your `setuptools`, using `pip install --upgrade setuptools`

**Reply by rokm** [reply]
I think this is a bug in the hook, nevertheless. The hook should not be excluding those modules, even if they are for outdated versions of python, because as indicated from above traceback, that old version of `setuptools` attemtps to import them from its other modules.

**Reply by bwoodsend** [reply]
Damn so it is. What on earth is that exclusion doing there...
